src/d/semantic/caster.d

```diff
@@ -336,7 +351,26 @@ struct Caster(bool isExplicit, alias bailoutOverride = null) {
 				}
 
 			case Float, Double, Real:
-				assert(0, "Floating point casts are not implemented");
+				// cast from a float to a float
+				if (isFloat(bt)) {
+					assert(bt != t);
+					return bt < t ? CastKind.FloatTrunc : CastKind.FloatExtend;
+				}
+
+				if (isExplicit) {
+					if (isIntegral(bt) || canConvertToIntegral(bt)) {
```

Comment:
The famous if/return/else pattern in action.

```d
if (!isIntegral(bt) && !canConvertToIntegral(bt)) {
  return CastKind.Invalid;
}

// Do the rest
```

Reply:
That's a habit from dmd style being to not have a leading ! in the first if statement condition.

Also so I could have the note in the bottom (weird void cast)

Reply:
The if statement is now the right way round